src/main/java/net/sf/jabref/gui/worker/CitationStyleToClipboardWorker.java

```diff
@@ -0,0 +1,193 @@
+                        processText(citations);
+                        break;
+                }
+            }
+
+            basePanel.frame().setStatus(Localization.lang("Copied %0 citations.", String.valueOf(selectedEntries.size())));
+        } catch (InterruptedException | ExecutionException e) {
+            LOGGER.error("Error while copying citations to the clipboard", e);
+        }
+    }
+
+    /**
+     * Generates a plain text string out of the preview and copies it additionally to the html to the clipboard
+     * (WYSIWYG Editors use the HTML, plain text editors the text)
+     */
+    private void processPreview(List<String> citations){
+        String html = "";
+        String plain = "";
+        for (String citation : citations) {
+            html += citation + "<br>";
+
+            String tmp = WHITESPACE.matcher(citation).replaceAll("");
+            tmp = REMOVE_HTML.matcher(tmp).replaceAll("");
+            plain += HTML_NEWLINE.matcher(tmp).replaceAll(OS.NEWLINE) + OS.NEWLINE + OS.NEWLINE;
+        }
+        new ClipBoardManager().setTransferableClipboardContents(new HtmlTransferable(html, plain));
+    }
+
+    /**
+     * Joins every citation with a newline and returns it.
+     */
+    private void processText(List<String> citations) {
+        String text = String.join("", citations);
+        new ClipBoardManager().setClipboardContents(text);
+    }
+
+    /**
+     * Converts the citations into the RTF format.
+     */
+    private void processRtf(List<String> citations) {
+        String result = "{\\rtf" + OS.NEWLINE + String.join("\\line ", citations) + "}";;
+        new ClipBoardManager().setTransferableClipboardContents(new RtfTransferable(result));
+    }
+
+    /**
+     * Inserts each citation into a XLSFO body and copies it to the clipboard
+     */
+    private void processXslfo(List<String> citations) {
+        String result = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" + OS.NEWLINE +
+                "<fo:root xmlns:fo=\"http://www.w3.org/1999/XSL/Format\">" + OS.NEWLINE +
+                "   <fo:layout-master-set>" + OS.NEWLINE +
+                "      <fo:simple-page-master master-name=\"citations\">" + OS.NEWLINE +
+                "         <fo:region-body/>" + OS.NEWLINE +
+                "      </fo:simple-page-master>" + OS.NEWLINE +
+                "   </fo:layout-master-set>" + OS.NEWLINE +
+                "   <fo:page-sequence master-reference=\"citations\">" + OS.NEWLINE +
+                "      <fo:flow flow-name=\"xsl-region-body\">" + OS.NEWLINE + OS.NEWLINE;
+
+        for (String citation : citations) {
+            result += citation + OS.NEWLINE;
+        }
+
+        result +="      </fo:flow>" + OS.NEWLINE +
+                "   </fo:page-sequence>" + OS.NEWLINE +
+                "</fo:root>" + OS.NEWLINE;
+
+        new ClipBoardManager().setTransferableClipboardContents(new HtmlTransferable(result, result));
```

Comment:
Could you update `HtmlTransferable` to offer `application/xml` as additional DataFlavor, which also returns the html? Maybe, a new class `XmlTransferable`? - I think, the whole `HtmlTransferable` has to be rewritten: `FLAVORS` as List and `getTransferDataFlavors()` using `toArray()`. `isDataFlavorSupported` using streaming.